include/zenoh-pico/collections/string.h

```diff
@@ -68,12 +68,12 @@ typedef struct {
 
 _z_string_t _z_string_null(void);
 _Bool _z_string_check(const _z_string_t *value);
-_z_string_t _z_string_make(const char *value);
-_z_string_t _z_string_n_make(const char *value, size_t len);
-_z_string_t _z_string_from_str(const char *value);
-_z_string_t _z_string_from_substr(const char *value, size_t len);
-_z_string_t _z_string_from_str_custom_deleter(char *value, _z_delete_context_t c);
-_z_string_t *_z_string_make_as_ptr(const char *value);
+_z_string_t _z_string_copy_from_str(const char *value);
+_z_string_t _z_string_copy_from_substr(const char *value, size_t len);
+_z_string_t *_z_string_copy_from_str_as_ptr(const char *value);
+_z_string_t _z_string_alias_str(const char *value);
+_z_string_t _z_string_alias_substr(const char *value, size_t len);
+_z_string_t _z_string_alias_str_custom_deleter(char *value, _z_delete_context_t c);
```

Comment:
this is not really an alias, it is sort of ownership transfer, so maybe the name should be different _z_string_move_from_str or something, and same for slice.